### PR TITLE
Update CI

### PR DESCRIPTION
**Pull request by Karthiccc23**
Update Android Actions CI.

- Issue - https://github.com/facebook/react-native/issues/35210

### CODE REVIEW

File: android/build.gradle [hunk 1]
```diff
@@ -13,7 +13,16 @@ buildscript {
     }
 }
 
+def REACT_NATIVE_VERSION = new File(['node', '--print',"JSON.parse(require('fs').readFileSync(require.resolve('react-native/package.json'), 'utf-8')).version"].execute(null, rootDir).text.trim())
```
Comment (friederbluemle) [inline android/build.gradle:16]: Same thing using Groovy native JsonSlurper:
```
def REACT_NATIVE_VERSION = new groovy.json.JsonSlurper().parse(new File("$rootDir/../node_modules/react-native/package.json")).version
```
No need to run an external problem (Node.js) here, plus the line becomes significantly shorter too.

File: android/build.gradle [hunk 2]
```diff
@@ -13,7 +13,16 @@ buildscript {
...
+
 allprojects {
+    configurations.all {
+        resolutionStrategy {
+            // Remove this override in 0.65+, as a proper fix is included in react-native itself.
+            force "com.facebook.react:react-native:" + REACT_NATIVE_VERSION
+        }
```
Comment (friederbluemle) [inline android/build.gradle:23]: I've seen this workaround, and it works well for applications projects - My only minor concern is this is a library project, and this closure is not conditionally scoped. Will this also affect consuming application projects? If not, then good to merge :+1:

**Review by friederbluemle — APPROVED** [review]
Thanks :+1:

File: android/build.gradle (hunk 1 above)
Comment (Karthiccc23) [reply android/build.gradle:16]: sure.

File: android/build.gradle (hunk 2 above)
Comment (Karthiccc23) [reply android/build.gradle:23]: it shouldn't affect the consuming projects.

Comment (friederbluemle) [reply android/build.gradle:23]: Sounds good, thanks!

**Pull request merged**